### PR TITLE
Optimizes conversion of bytes to words

### PR DESCRIPTION
**Pull request by lordshashank**
## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Resolves: #940 

## What is the new behavior?

- Introduces new function `bytes_32_words_size` which return number of 32 byte words required to represent `size`
- Replaces the `ceil32` function with `bytes_32_word_size` wherever applicable.

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

- - -
This change is [Reviewable](https://reviewable.io/reviews/kkrt-labs/kakarot-ssj/985)

### CODE REVIEW

File: crates/evm/src/create_helpers.cairo [hunk 1]
```diff
@@ -48,7 +48,7 @@ pub impl CreateHelpersImpl of CreateHelpers {
         let charged_gas = match create_type {
             CreateType::Create => gas::CREATE + memory_expansion.expansion_cost + init_code_gas,
             CreateType::Create2 => {
-                let calldata_words = ceil32(size) / 32;
+                let calldata_words = bytes_32_words_size(size) / 32;
```
Comment (enitrat) [inline crates/evm/src/create_helpers.cairo:51]:
```suggestion
                let calldata_words = bytes_32_words_size(size);
```

you have one extra `/32` here

File: crates/evm/src/gas.cairo [hunk 2]
```diff
@@ -194,7 +194,7 @@ pub fn memory_expansion(
         }
     }?;
 
-    let new_size = helpers::ceil32(max_size);
+    let new_size = helpers::bytes_32_words_size(max_size) * 32;
```
Comment (enitrat) [inline crates/evm/src/gas.cairo:197]: for this specific case, we want the size in _bytes_, not in 32-bytes words. Thus we can keep using ceil32

```suggestion
    let new_size = helpers::ceil32(max_size);
```

File: crates/utils/src/helpers.cairo [hunk 3]
```diff
@@ -55,6 +55,22 @@ pub fn ceil32(value: usize) -> usize {
...
+/// The number of 32-byte words required to represent `size` bytes
+///
+/// # Examples
+/// bytes_32_words_size(2) = 1
+/// bytes_32_words_size(34) = 2
+#[inline(always)]
+pub fn bytes_32_words_size(size: usize) -> usize {
+    (size + 31) / 32
+}
+
```
Comment (enitrat) [inline crates/utils/src/helpers.cairo:73]: what happens if `size + 31` > MAX_USIZE ?

In that case, the function will panic. What we can do to avoid this is use the `SaturatingAdd` trait to saturate the result to the max size of a usize.

Also, please add a test for this edge case

File: crates/evm/src/instructions/environmental_information.cairo [hunk 4]
```diff
@@ -238,8 +238,8 @@ pub impl EnvironmentInformationImpl of EnvironmentInformationTrait {
         }
         ensure(!(last_returndata_index > return_data.len()), EVMError::ReturnDataOutOfBounds)?;
 
-        //TODO: handle overflow in ceil32 function.
-        let words_size = (ceil32(size.into()) / 32).into();
+        //TODO: handle overflow in bytes_32_words_size function.
```
Comment (enitrat) [inline crates/evm/src/instructions/environmental_information.cairo:241]:
```suggestion
```

Once handled in `bytes_32_words_size` we can remove this TODO

File: crates/utils/src/helpers.cairo (hunk 3 above)
Comment (enitrat) [reply crates/utils/src/helpers.cairo:73]: we can also delete ceil32 if it's not longer used anywhere :)

**Review by enitrat — APPROVED** [review]
